src/components/SurveyAnswer/ChoiceAnswer/MultiplePickAnswer.tsx

```diff
@@ -0,0 +1,74 @@
+import Image from 'next/image';
+import SimpleBar from 'simplebar-react';
+
+import { SurveyAnswerProps } from '..';
+
+import styles from './ChoiceAnswer.module.scss';
+import 'simplebar-react/dist/simplebar.min.css';
+
+const MultiplePickAnswer = ({
+  question,
+  currentResponse,
+  onResponseChange,
+}: SurveyAnswerProps) => {
+  const onAnswerSelect = (answerId: string) => {
+    const currentAnswers = currentResponse?.answers || [];
+    const isAnswerSelected = currentAnswers.some(({ id }) => id === answerId);
+
+    if (isAnswerSelected) {
+      const updatedAnswers = currentAnswers.filter(({ id }) => id !== answerId);
+
+      if (updatedAnswers.length === 0) {
+        return onResponseChange(null);
+      }
+
+      onResponseChange({
+        questionId: question.id,
+        answers: updatedAnswers,
+      });
+    } else {
+      onResponseChange({
+        questionId: question.id,
+        answers: currentAnswers.concat({ id: answerId }),
+      });
+    }
+  };
+
+  const answerStateClass = (answerId: string) => {
+    const isActiveAnswer = currentResponse?.answers.some(
+      ({ id }) => id === answerId
+    );
+
+    if (isActiveAnswer) {
+      return styles.activeAnswer;
+    }
+  };
```

Comment:
I'm not sure if it is the case for the other types of Answers, but I just have a though about performence and re-rendering... 🤔 

As we are reaching the `currentResponse` component property to get the `currentAnswers`, it means that when a user select 1 answer, the following happened:
- `onAnswerSelect` method is called.
- `onResponseChange` is called and will update the value of `currentReponse.currentAnswers`
- The value update will trigger the re-rendering of the `MultiplePickAnswer` component (which contain a foreach look across all the possible answers....).

Did you use React Dev Tools (Profiler) to observer how components get re-rendered? (as I'm just assuming the setps above). 
The best-case scenario would be to **only** rerender the selected answer when it is clicked. 

This might require to extract the `answer` component from the `picker` component, so that it can get its own state and the `isActiveAnswer` will not need to itterate across the list of `currentResponse.anwers` (except for the initial state that would be a boolean property on that component) 🤔

Reply:
I used the profiler for this component and found that when an answer is selected, its re-renders at the page level (`Question` page). So to fix the re-render issue is beyond this component level, and I should make a different PR or task on that.

## Profiler on select answer action

https://user-images.githubusercontent.com/14077479/181052277-98dcb256-5f52-4499-9ee8-9aab1c62c63a.mov

## Profiler report

<img width="666" alt="image" src="https://user-images.githubusercontent.com/14077479/181052502-7b115572-e8c1-4924-a766-64869e62ac5f.png">

Regarding the concern that you mentioned about iterating through the list of `currentResponse.anwers` to determine active answers. I separate the MultiSelectAnswerItem into its own component, and its active status relies on the internal state so that when a user selects/de-selects the answer, it doesn't need to go through the list anymore.

I updated in [6303f6d](https://github.com/hanam1ni/react-surveyor/pull/97/commits/6303f6df5aeb51d41139636e111a04bbe5c29d5f)

Reply:
Nice!! Thanks for your detailed answer. 💯  for not implementing the whole refactoring in this PR, it would be too much ^^